src/toil_scripts/adam_uberscript/adam_uberscript.py

```diff
@@ -0,0 +1,457 @@
+                                   'toil-leader',
+                                   '-o', 'UserKnownHostsFile=/dev/null',
+                                   '-o', 'StrictHostKeyChecking=no',
+                                   'screen', '-S', params.cluster_name,
+                                   '-X', 'stuff', '"{0}"'.format(chunk)])
+    
+    except subprocess.CalledProcessError as e:
+        log.info('Pipeline exited with non-zero status code: {}'.format(e))
+
+
+def get_metric(cw, metric, instance_id, start, stop):
+    """
+    returns metric object associated with a paricular instance ID
+
+    metric_name: str            Name of Metric to be Collected
+    instance_id: str            Instance ID
+    start: float                ISO format of UTC time start point
+    stop: float                 ISO format of UTC time stop point
+    :return: metric object
+    """
+    namespace, metric_name = metric.rsplit('/', 1)
+    metric_object = cw.get_metric_statistics(namespace=namespace,
+                                             metric_name=metric_name,
+                                             dimensions={'InstanceId': instance_id},
+                                             start_time=start,
+                                             end_time=stop,
+                                             period=300,
+                                             statistics=['Average'])
+    return metric_object
+
+
+def get_cluster_size(cluster_name):
+    """
+    Returns the number of running toil-worker nodes
+    """
+    numlines = subprocess.check_output(["cgcloud",
+                                        "list",
+                                        "-c", cluster_name, "toil-worker"]).count("\n")
+    return numlines - 1 # Subtract 1 for header
+
+
+def get_desired_cluster_size(conn, dom):
+    
+    nodes_per_sample = Samples.load(conn, dom)
+    return sum(map(lambda x: x[1], nodes_per_sample.samples.values()))
+
+
+def update_cluster_size(conn, dom, n):
+    ClusterSize.change_size(conn, dom, n)
+
+
+def grow_cluster(n, instance_type, cluster_name, spot_bid):
+    """
+    grow the cluster by n nodes
+    """
+
+    old_size = get_cluster_size(cluster_name)
+    target_size = old_size + n
+    diff = n
+
+    # loop until we've grown the cluster by our target size
+    while diff > 0:
+        log.info('Growing cluster by {} nodes of type: {}'.format(diff, instance_type))
+
+        # call to cgcloud to increase the cluster size
+        subprocess.check_call(['cgcloud',
+                               'grow-cluster',
+                               '--list',
+                               '--instance-type', str(instance_type),
+                               '--num-workers', str(diff),
+                               '--cluster-name', str(cluster_name),
+                               '--spot-bid', str(spot_bid),
+                               '--zone', ("%sa" % aws_region),
```

Comment:
Big piece of functionality missing here. We need to balance between the availability zones.

I'll file a PR against this one.